### DISCUSSION TITLE
How to use `is_key_content_base64` for `Spaceship`

### DISCUSSION THREAD

**Discussion by mathaeus** [fastlane/fastlane#18009] [Q&A]
Hello, from what I see it's not possible to call:

`Spaceship::ConnectAPI.auth(...)` and provide the `is_key_content_base64` that was added in the PR below for calling it via `app_store_connect_api_key`.
https://github.com/fastlane/fastlane/pull/17401/files

We are injecting the value for the key via ENV variables and keep getting the error that was also mentioned here https://github.com/fastlane/fastlane/issues/17340 if the content is not base64-encoded.
I actually tried it locally with setting the `is_key_content_base64` to `true` and then things work as expected.

ℹ️ We use some "raw" `Spaceship` API calls for some things in our workflows and are currently migrating to use the Auth Key approach, and therefore would need a solution there.

**So long story short, either I'm missing something or this feature of providing the `is_key_content_base64` is missing. Please let me know what you think.** @joshdholtz (sorry for pinging) or anyone else.

**Comment by mathaeus** [comment]
From what I see I can directly use 
```
Spaceship::ConnectAPI.token = Spaceship::ConnectAPI::Token.create(app_store_connect_api_key)
```
 instead of calling 
```
Spaceship::ConnectAPI.auth(...)
```

but I'm not sure if this is a proper solution 🤔

**Comment by dinsen** [comment]
Hi @mathaeus 
I'm setting the token by my self.
```ruby
token = Spaceship::ConnectAPI::Token.from_json_file('authkey.json')
Spaceship::ConnectAPI.token = token
```
I then use the fastlane auth key json file
https://docs.fastlane.tools/app-store-connect-api/#using-fastlane-api-key-json-file

Inside that you can specify the is_key_content_base64.

Will that help on your problem?

**Reply by mathaeus** [reply]
Hi @Dinsen,

thanks for your answer! As I posted above I was able to make it work as well with setting the token directly.

As you can see I'm calling
```
Spaceship::ConnectAPI.token = Spaceship::ConnectAPI::Token.create(app_store_connect_api_key)
```

and `app_store_connect_api_key` has the API key + base64 ENV variables set.

I was just not sure if this approach is a good one in the end, since I would prefer using the dedicated `auth` method.

**Comment by mathaeus** [comment]
In case anyone else is trying to figure this out for Ruby 3 as well, in order to pass the hash that you get from calling `app_store_connect_api_key` to `Token.create`, with Ruby 3 you need to use the double splat operator `**`, see [here](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/).

```
Spaceship::ConnectAPI.token = Spaceship::ConnectAPI::Token.create(**app_store_connect_api_key)
```
